custom_components/fellow/fellow_aiden/__init__.py

```
@@ -194,287 +255,312 @@ def __device(self):
         match = re.search(pattern, link)
         if not match:
             raise ValueError("Invalid profile URL or ID format")
         brew_id = match.group(1)
-        self._log.debug("Brew ID: %s" % brew_id)
+        self._log.debug("Brew ID: %s", brew_id)
         shared_url = self.BASE_URL + self.API_SHARED_PROFILE.format(bid=brew_id)
-        response = self._request_with_reauth("get", shared_url)
-        if response.status_code == 404:
+        response = await self._request_with_reauth("get", shared_url)
+        if response.status == 404:
             raise ValueError(f"Failed to fetch profile (ID: {brew_id})")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)